### PR TITLE
More useful browser test progress

### PR DESCRIPTION
**Pull request by nanaya**
Use `dataProvider` instead of running everything as single test and handling errors manually.

- [x] depends on #7670

### CODE REVIEW

File: phpunit.dusk.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
          convertErrorsToExceptions="true"
          convertNoticesToExceptions="true"
          convertWarningsToExceptions="true"
-         processIsolation="true"
```
Comment (nanaya) [inline phpunit.dusk.xml:9]: @nekodex, do you remember why this is set to `true`? Checking [the dusk template](https://github.com/laravel/dusk/blame/6.x/stubs/phpunit.xml) looks like it's always `false`. Is it some leftover from experiments?

File: tests/Browser/SanityTest.php [hunk 2]
```diff
@@ -180,76 +58,276 @@ public function cleanup()
...
+    }
+
+    private static function filterLog(array $log)
+    {
+        $return = [];
+
+        foreach ($log as $line) {
+            if ($line['source'] === 'network') {
+                $matches = [];
+                $count = preg_match_all("/^([^ ]+) - Failed to load resource: the server responded with a status of ([0-9]{3}) \(([^\)]*)\)$/i", $line['message'], $matches);
```
Comment (notbakaneko) [inline tests/Browser/SanityTest.php:203]: why is this `$count` even here 🤔

File: phpunit.dusk.xml (hunk 1 above)
Comment (nekodex) [reply phpunit.dusk.xml:9]: I have a feeling it was to try and work-around something weird but I don't remember exactly what, sorry.

If setting it to false doesn't cause any issues now, it's probably fine to change?

Comment (notbakaneko) [reply phpunit.dusk.xml:9]: I'm actually getting some weirdness where the test sometimes runs as the wrong user causing the expectations to fail but I haven't been able to isolate it yet 😕

**Review by notbakaneko — CHANGES_REQUESTED** [review]
- Would be useful to add `app('groups')->resetCache()` before the browser tests start (or have the cache stored in different namespace)
- `run_dusk.sh` should be updated to pass flags to `phpunit`

**Review by notbakaneko — APPROVED** [review]

**Pull request merged**